test-support/page-object/custom-helper.js

```diff
@@ -0,0 +1,30 @@
+import Ember from 'ember';
+import { build } from './build';
+import { qualifySelector } from './helpers';
+
+export function customHelper(userDefinedFunction) {
+  return function(selector, options) {
+    // def
+    return {
+      buildPageObjectAttribute: function(key, parent) {
+        // def
+        return function() {
+          // eval
+          let scopedSelector = qualifySelector(parent.scope, selector);
+
+          let customHelperRes = userDefinedFunction(scopedSelector, options);
```

Comment:
Maybe we can simplify the code by renaming some variables `s/customHelperRes/response` `s/scopedSelector/qualifiedSelector/`

Also, maybe we can improve readability by having only one return statement

```js
if ($.isPlainObject(response)) {
    processedResponse = build(...);
} else if ($.isFunction(response)) {
    processedResponse = response();
} else {
    processedResponse = response;
}

return processedResponse;
```

What do you think?